src/include/self_driving/forecast/workload_forecast.h

```diff
@@ -25,8 +28,16 @@ class WorkloadForecast {
    */
   explicit WorkloadForecast(uint64_t forecast_interval);
 
+  /**
+   * Get number of forecasted segments
+   * @return number of forecasted segments
+   */
+  uint64_t GetNumberOfSegments() { return num_forecast_segment_; }
+
  private:
   friend class PilotUtil;
+  friend class Pilot;
+  friend class pilot::MonteCarloTreeSearch;
```

Comment:
You can still keep the friend class if you want to restrict the Getters/Setters to a few specific classes, but don't use friend class as a way to access private members.